src/k2/transport/RPCDispatcher.cpp

```diff
@@ -150,43 +157,62 @@ void RPCDispatcher::_handleNewMessage(Request&& request) {
     }
 }
 
-void RPCDispatcher::_send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta) {
 
+seastar::future<>
+RPCDispatcher::_send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta) {
+    auto ep = RPC().getServerEndpoint(endpoint.getProtocol());
     auto protoi = _protocols.find(endpoint.getProtocol());
     if (protoi == _protocols.end()) {
         K2WARN("Unsupported protocol: "<< endpoint.getProtocol());
-        return;
+        return seastar::make_ready_future<>();
     }
     auto serverep = protoi->second->getServerEndpoint();
     K2DEBUG("sending message for verb: " << int(verb) << ", to endpoint=" << endpoint.getURL()
-            << ", with server endpoint: " << (serverep ? serverep->getURL() : String("none")));
-    if (serverep && endpoint == *serverep) {
-        // deliver via a future to possibly yield if there is a loop of send/receive requests
-        (void) seastar::sleep(0ns)
-            .then([disp=weak_from_this(), verb, endpoint, meta=std::move(meta), payload=std::move(payload)] () mutable {
-                if (disp) {
-                    // rewind the payload to the correct position
```

Comment:
please retain this comment as it explains why we're modifying the payload cursor

Reply:
Put this comment back.